.github/workflows/main.yaml

```diff
@@ -0,0 +1,312 @@
+        mkdir ${context} && cd ${context} 
+        echo "FROM ${BASE_IMAGE}" > Dockerfile
+        echo "RUN chown 1001:root /usr/bin/caddy" >> Dockerfile
+      env:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        SOURCE_IMAGE_REGISTRY: ${{ matrix.SOURCE_IMAGE_REGISTRY }}
+        SOURCE_IMAGE_NAME: ${{ matrix.SOURCE_IMAGE_NAME }}
+        SOURCE_IMAGE_TAG: ${{ matrix.SOURCE_IMAGE_TAG }}
+        BASE_SOURCE_IMAGE_REGISTRY: ${{ matrix.BASE_SOURCE_IMAGE_REGISTRY }}
+        BASE_SOURCE_IMAGE_NAME: ${{ matrix.BASE_SOURCE_IMAGE_NAME }}
+        BASE_SOURCE_IMAGE_TAG: ${{ matrix.BASE_SOURCE_IMAGE_TAG }}
+        REGISTRY_USERNAME: ${{ secrets.ARTIFACTORY_USERNAME }}
+        REGISTRY_PASSWORD: ${{ secrets.ARTIFACTORY_PASSWORD }}
+
+    - name: Log in to the GHCR
+      uses: docker/login-action@v3
+      with:
+        registry: ghcr.io
+        username: ${{ github.actor }}
+        password: ${{ secrets.GITHUB_TOKEN }} 
+
+    - name: Prepare docker tags for image
+      id: meta
+      if: contains(fromJSON('["aries-endorser-db","aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+      uses: docker/metadata-action@v5
+      with:
+        images: ghcr.io/bcgov/dts-endorser-service/${{ matrix.service }}
+        flavor: |
+          latest=true
+        tags: |
+          type=schedule
+          type=ref,event=branch
+          type=ref,event=pr
+          type=semver,pattern={{version}}
+          type=semver,pattern={{major}}.{{minor}}
+          type=semver,pattern={{major}}
+          type=sha,value=latest     
+        annotations: |
+          org.opencontainers.image.description="Hyperledger Aries Cloud Agent Python (ACA-Py) is a foundation for building decentralized id"
+          
+    - name: Update Docker base image and pull the base image if access is required 
+      if:   matrix.BASE_SOURCE_IMAGE_REGISTRY  != '' && contains(fromJSON('["aries-endorser-db","aries-endorser-agent","aries-endorser-backup","aries-endorser-api","aries-endorser-proxy"]'), matrix.service)
+      run: |  
+        if [[ -n "${REGISTRY_USERNAME}" ]]; then
+          docker login -u  $REGISTRY_USERNAME -p $REGISTRY_PASSWORD $BASE_SOURCE_IMAGE_REGISTRY
+        fi
+        BASE_IMAGE="${BASE_SOURCE_IMAGE_REGISTRY}${BASE_SOURCE_IMAGE_NAME}:${BASE_SOURCE_IMAGE_TAG}"
+        sed -i -e "s;FROM .*;FROM ${BASE_IMAGE};g" "$file"
+        docker pull $BASE_IMAGE
+      env:
+        context: ${{ matrix.SOURCE_CONTEXT_DIR }}
+        BASE_SOURCE_IMAGE_REGISTRY: ${{ matrix.BASE_SOURCE_IMAGE_REGISTRY }}
+        BASE_SOURCE_IMAGE_NAME: ${{ matrix.BASE_SOURCE_IMAGE_NAME }}
+        BASE_SOURCE_IMAGE_TAG: ${{ matrix.BASE_SOURCE_IMAGE_TAG }}
+        file: ${{ matrix.SOURCE_CONTEXT_DIR }}/${{ matrix.DOCKER_FILE_PATH }}
+        REGISTRY_USERNAME: ${{ secrets[matrix.REGISTRY_USERNAME_SECRET_NAME]}}
+        REGISTRY_PASSWORD: ${{ secrets[matrix.REGISTRY_PASSWORD_SECRET_NAME]}}
```

Comment:
The docker build action used the parameters passed to login to docker before building the image.
However, we need to pull the image locally because the docker build action will re-login to ghcr, that will provoke the access to pull the base image.